src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/JsonFormatWriterGenerator.cs

```diff
@@ -98,9 +103,11 @@ internal JsonFormatCollectionWriterDelegate GenerateCollectionWriter(CollectionD
                 return (JsonFormatCollectionWriterDelegate)_ilg.EndMethod();
             }
 
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2070:UnrecognizedReflectionPattern",
```

Comment:
Can we add a new helper/utility method named `GetInvokeMethod(Type delegateType)` that suppresses this message and does the `Debug.Assert`? That way we are only suppressing a single GetMethod call, and aren't over suppressing other warnings.